server/broadcast.go

```diff
@@ -207,11 +207,31 @@ func removeSessionFromList(sessions []*BroadcastSession, sess *BroadcastSession)
 
 func selectSession(sessions []*BroadcastSession, exclude []*BroadcastSession, durMult int) *BroadcastSession {
 	for _, session := range sessions {
-		if len(session.SegsInFlight) > 0 &&
-			time.Since(session.SegsInFlight[0].startTime) < time.Duration(durMult)*session.SegsInFlight[0].segDur &&
-			!includesSession(exclude, session) {
-			// Re-use last session if oldest segment is in-flight for < durMult * segDur
-			return session
+		// A session in the exclusion list is not selectable
+		if includesSession(exclude, session) {
```

Comment:
Even though it's a side effect of your change, I really like the readability of having these broken out into distinct checks

Reply:
Glad you agree that this is more readable! TBH I was having trouble parsing the one-liner with multiple logical operators so felt compelled to do the small refactor.